### PR TITLE
[WIP] Improve shipyard commands

### PR DESCRIPTION
**Pull request by Itxaka**

### CODE REVIEW

**Comment by jsuchome** [comment]
I'm not that sure about the retry part. Aren't we able to fix the problem?

**Comment by Itxaka** [comment]
> I'm not that sure about the retry part. Aren't we able to fix the problem?

Im not sure of what even the issue is. If its an internal temp routing error, sure we may be able to, if its  a locking issues, we may be able to. 

In any case, this is missing the WIP tag as I want to see if I can hit the issue so the failed_when works correctly :)

File: playbooks/roles/airship-deploy-osh/tasks/main.yml [hunk 1]
```diff
@@ -134,7 +137,8 @@
     SHIPYARD_IMAGE: "{{ shipyard_image }}"
     OS_PASSWORD: "{{ lookup('password', secrets_location + '/ucp_shipyard_keystone_password ' + password_opts) }}"
   register: shipyard_update_software
-  failed_when: shipyard_update_software.stdout.find('Error') == 0
+  failed_when:
+    - shipyard_update_software.stdout.find('Errors: 0') != 0
```
Comment (evrardjp) [inline playbooks/roles/airship-deploy-osh/tasks/main.yml]: you should wrap this with double quotes like above.

Comment (Itxaka) [reply playbooks/roles/airship-deploy-osh/tasks/main.yml]: argh

**Comment by Itxaka** [comment]
umm doesnt seem to be working, migth need to regex it. With this output the task succeeded but it should not have to.

```
[2019-09-10T10:06:45.825Z]     Status: Downloaded newer image for quay.io/airshipit/shipyard:a9ff859df980fbb95a912fc49312ef3c51f65b3f-opensuse_15
[2019-09-10T10:06:45.825Z]   stderr_lines: <omitted>
[2019-09-10T10:06:45.825Z]   stdout: |-
[2019-09-10T10:06:45.825Z]     Error: Unhandled Exception raised:
[2019-09-10T10:06:45.825Z]     Reason:
[2019-09-10T10:06:45.825Z]     - Error: An error occurred, but was not specified
[2019-09-10T10:06:45.825Z]   
[2019-09-10T10:06:45.825Z]     ####  Errors: 1, Warnings: 0, Infos: 0, Other: 0  ####
[2019-09-10T10:06:45.825Z]   stdout_lines: <omitted>
```

File: playbooks/roles/airship-deploy-osh/tasks/main.yml (hunk 1 above)
Comment (evrardjp) [reply playbooks/roles/airship-deploy-osh/tasks/main.yml]: and probably add a different condition :)

**Comment by evrardjp** [comment]
> umm doesnt seem to be working, migth need to regex it. With this output the task succeeded but it should not have to.
> 
> ```
> [2019-09-10T10:06:45.825Z]     Status: Downloaded newer image for quay.io/airshipit/shipyard:a9ff859df980fbb95a912fc49312ef3c51f65b3f-opensuse_15
> [2019-09-10T10:06:45.825Z]   stderr_lines: <omitted>
> [2019-09-10T10:06:45.825Z]   stdout: |-
> [2019-09-10T10:06:45.825Z]     Error: Unhandled Exception raised:
> [2019-09-10T10:06:45.825Z]     Reason:
> [2019-09-10T10:06:45.825Z]     - Error: An error occurred, but was not specified
> [2019-09-10T10:06:45.825Z]   
> [2019-09-10T10:06:45.825Z]     ####  Errors: 1, Warnings: 0, Infos: 0, Other: 0  ####
> [2019-09-10T10:06:45.825Z]   stdout_lines: <omitted>
> ```

Find doesn't take regexps. There is an ansible filter to search with regexp, you might want to do that on your stdout instead.

**Pull request closed without merging**